app/utils/SentTxStore.js

```diff
@@ -3,8 +3,26 @@ import path from 'path';
 import fs from 'fs';
 import { remote } from 'electron';
 import { Transaction, TxDetail } from '../components/AppState';
+import parseMemo from './parseMemo';
 
 export default class SentTxStore {
+  static async writeSentTx(sentTx) {
+    const fileName = await this.locateSentTxStore();
+
+    if (!fs.existsSync(fileName)) {
+      await fs.promises.writeFile(fileName, JSON.stringify([sentTx]));
+      return;
+    }
+
+    fs.readFile(fileName, async (err, data) => {
```

Comment:
should this be `await fs.promises.readFile(...`?

Reply:
no, don't think so. we're not waiting for `fs.readFile` itself. we invoke the function, which is an async function, it gets added to the microtask queue to be executed when the call stack is empty, and once it runs then our async callback is invoked. we're not dependant on it anywhere else so we can just queue it up 👍